src/exceptions/utils.ts

```diff
@@ -84,8 +82,8 @@ export const throwExpected = async <T = any>(fn: () => Promise<T>, exceptionsMap
 
     return result;
   } catch (e: unknown) {
-    const response = toPlatformErrorResponse((e as any)?.response);
-    const errCode = response?.body.statusCode ?? parseHTTPErrorCode(e) ?? FALLBACK_HTTP_ERROR_CODE;
+    const response = await toPlatformErrorResponse((e as any)?.response);
```

Comment:
`toPlatformErrorResponse` seems expecting error/exception `(e: unknown)` but we are passing response to it?

Reply:
The old code was a bit misleading, I improved the comment and the typing

Reply:
The purpose of `toPlatformErrorResponse` is to provide information about error responses for processing exceptions. Basically, when you get 404 error and you know that it might mean several different things (e.g. cluster or space not being found when fetching cluster info from a space), - you need some way to parse the response to distinguish between these cases. This function wraps server error responses in a typed interface, makes it simpler to handle. Having it paired with types from axios makes it better IMO.